### ISSUE TITLE
Ambiguous behaviour in tests when using the $request->user() with custom guards

### ISSUE THREAD

**Issue by barnabaskecskes** [laravel/framework#31506]
- Laravel Version: 6.15.1
- PHP Version: 7.3.10
- Database Driver & Version: SQLite 3.28.0

### Description:
Assume a custom `admin` guard operating on the `admins` table. When you write a test such as 
```
$response = $this->actingAs($admin, 'admin')->get(route('test'));
```
and you obtain the user from the request object in your controller, for example
```
request()->user()
```
within the test, you will get your `$admin` back as expected. However, when you're running your application via the browser for example, you will receive `null` regardless that you have your `$admin` authenticated, unless the route explicitly uses the `auth:admin` middleware.

For that reason I believe that the `actingAs($user, $guard)` method within the test _assumes_ the `auth:{$guard}` middleware to be used when running the requests and that might lead to unexpected/ambiguous behaviour.

> NOTE: I do understand that this happens because the guard not being specified results in using the fallback default and that is indeed the correct behaviour; what I'm trying to say is that the test behaviour is slightly different and it can potentially cause problems.

### Steps To Reproduce:
1) Clone the repository and install the dependencies
```
git clone https://github.com/barnabaskecskes/laravel-custom-guard-test-authentication-issue
cd laravel-custom-guard-test-authentication-issue
composer install
```
2) Run [the test](https://github.com/barnabaskecskes/laravel-custom-guard-test-authentication-issue/blob/master/tests/Feature/LoggingInThroughTheAdminGuardTest.php#L13-L22) to see it returning the admin user
```
vendor/bin/phpunit --filter LoggingInThroughTheAdminGuardTest
```
3) Serve the app and see the different behaviour in the browser (check [the view](https://github.com/barnabaskecskes/laravel-custom-guard-test-authentication-issue/blob/master/resources/views/welcome.blade.php#L13-L30) and [the routes file](https://github.com/barnabaskecskes/laravel-custom-guard-test-authentication-issue/blob/master/routes/web.php#L7-L16) to see what you should expect)
```
php artisan serve
```

**Comment by driesvints** [comment]
@barnabaskecskes you're correct. I managed to reproduce this. Appreciating any help with looking into this.

**Comment by barnabaskecskes** [comment]
@driesvints I know that if you take [this line](https://github.com/laravel/framework/blob/6.x/src/Illuminate/Foundation/Testing/Concerns/InteractsWithAuthentication.php#L36) out, it follows the correct behaviour _in this instance_ but I did not have a chance to dig any deeper than that just yet. I'll try to find some time to dissect this a bit more.

**Comment by driesvints** [comment]
@barnabaskecskes your solution from the pr might not be ideal but maybe we can solve this differently. I'll leave this open so it can be picked up eventually. Maybe someone has another idea.

**Comment by barnabaskecskes** [comment]
@driesvints I appreciate that, thank you.

I have saved that PR as a _Draft_ - wasn't even sure it was going to appear. All it said that I need to make it _Ready for review_ eventually - that didn't go down well. 😅 

I'm having a bit of a question here on how to reliably test things without relying on arbitrary tests from another project? (see PR comments) If somebody knows a way, please ping me!

**Comment by driesvints** [comment]
https://github.com/laravel/docs/pull/6007
